lms/djangoapps/experiments/flags.py

```diff
@@ -128,6 +135,13 @@ def get_bucket(self, course_key=None, track=True):
             # We need username for stable bucketing and id for tracking, so just skip anonymous (not-logged-in) users
             return 0
 
+        if is_masquerading(request.user, course_key) and is_masquerading_as_specific_student(request.user, course_key):
+            user = get_specific_masquerading_user(request.user, course_key)
```

Comment:
This can return None, but that isn't handled below and would cause an exception later at `user.username` or whatever.

Which is fine here, because you're re-encoding the None condition in the if above. But that's brittle in the case one of those `if` conditions change. What about:

```
user = get_specific_masquerade_user(request.user, course_key)
if user:
    masquerade_active = True
else:
    user = request.user
    masquerade_active = False
```

(Or some formulation that you think is prettier.)

Reply:
Yeah, that's cleaner.